pytorch_lightning/trainer/trainer.py

```diff
@@ -834,6 +869,63 @@ def __test_given_model(self, model, test_dataloaders):
 
         return results
 
+    def predict(
+        self,
+        model: Optional[LightningModule] = None,
+        dataloaders: Optional[Union[DataLoader, List[DataLoader]]] = None,
+    ):
+        r"""
+
+        Separates from fit to make sure you never run on your predictions set until you want to.
+
+        This will call the model forward function to compute predictions.
+
+        Args:
+            model: The model to predict on.
+
+            dataloaders: Either a single
+                Pytorch Dataloader or a list of them, specifying inference samples.
+
+        Returns:
+            The final test result dictionary. If no test_epoch_end is defined returns a list of dictionaries
```

Comment:
Can you make this more clear? like what's the key and value of this result dictionary? And I guess it's a list of dictionary if multiple dataloaders are passed? (I am not sure about this) `Trainer.test()` needs the same doc improvements, too.

Reply:
Thanks for the feedback. I will update this :)

Reply:
Returns a list of dictionaries, one for each test dataloader containing their respective metrics.